### PR TITLE
fix fd_in_limits test on Windows

### LINKED ISSUE #52506: CI failure in FileWatching
https://buildkite.com/julialang/julia-master/builds/30998#018c4b64-01de-458b-82ce-955c376eef94
```
FileWatching                                     (2) |         failed at 2023-12-08T22:56:13.849
Test Failed at C:\buildkite-agent\builds\win2k22-amdci6-0\julialang\julia-master\julia-57e6fe0e14\share\julia\stdlib\v1.11\FileWatching\test\runtests.jl:36
  Expression: fd_in_limits
```
This test appears to just be bad (the fd_in_limits should be typemax DWORD, not DWORD16). It failed on CI, but CI failed to report this issue.

### PR DESCRIPTION
**Pull request by stevengj**
Closes #52506.

`OS_HANDLE` on Windows is an alias for `WindowsRawSocket`, so I looked at the [win32 socket documentation](https://learn.microsoft.com/en-us/windows/win32/winsock/socket-data-type-2), which says that a socket handle "may take any value in the range `0` to `INVALID_SOCKET–1`."   The `INVALID_SOCKET` is [defined as `~0`](https://stackoverflow.com/questions/10817252/why-is-invalid-socket-defined-as-0-in-winsock2-h-c), i.e. `0xffffffff`.

But I'm not 100% sure the socket docs apply here?  @vtjnash's comment said it should be "in `typemax(DWORD)`" (`== typemax(UInt32)`), which would correspond to `0xffffffff` and not `0xffffffff - 1`.     But I can't find clear docs on the value of a `HANDLE` — if it is really just a pointer, couldn't it be just about anything? (The old code corresponds to `typemax(Int16) == 0x7fff`,  which suggests changing it to `typemax(Int32) == 0x7fffffff`, but I'm not sure where that is coming from either.)   Is there a relevant section of the Win32 docs I should be looking at?

Note that this test dates back to #26341 by @vtjnash.

### CODE REVIEW

File: stdlib/FileWatching/test/runtests.jl [hunk 1]
```diff
@@ -24,7 +24,7 @@ for i in 1:n
         uv_error("pipe", ccall(:uv_pipe, Cint, (Ptr{NTuple{2, Base.OS_HANDLE}}, Cint, Cint), Ref(pipe_fds, i), 0, 0))
     end
     Ctype = Sys.iswindows() ? Ptr{Cvoid} : Cint
-    FDmax = Sys.iswindows() ? 0x7fff : (n + 60 + (isdefined(Main, :Revise) * 30)) # expectations on reasonable values
+    FDmax = Sys.iswindows() ? 0xfffffffe : (n + 60 + (isdefined(Main, :Revise) * 30)) # expectations on reasonable values
     fd_in_limits =
         0 <= Int(Base.cconvert(Ctype, pipe_fds[i][1])) <= FDmax &&
```
Comment (stevengj) [inline stdlib/FileWatching/test/runtests.jl:29]: Shouldn't this be converting to `UInt` on Windows?   On 32-bit systems an `Int` seems wrong.

Comment (Keno) [reply stdlib/FileWatching/test/runtests.jl:29]: Yeah, doing this comparison in UInt seems right.

**Comment by stevengj** [comment]
(Even on Unix, having a test for a file descriptor in a "reasonable range" kind of scares me.   Should we just delete this test entirely?  It's already disabled on Macs.)

**Comment by vtjnash** [comment]
I am not certain where the typemax(Int16) came from there. The typemax(Int32) comes from the (undocumented) fact that the kernel observably internally truncates these handles to 32 bits, and just ignores the high bits, but really it should be legal to treat it as a signed value as well, so it is `typemax(Int32)` not `typemax(UInt32)-1`

**Comment by stevengj** [comment]
> but really it should be legal to treat it as a signed value as well, so it is `typemax(Int32)` not `typemax(UInt32)-1`

Why are negative values not allowed?

**Comment by vtjnash** [comment]
I think the kernel uses those for error codes, but also just because they can be unexpected to encounter, so the kernel probably avoids them.

**Comment by stevengj** [comment]
Okay, I’ve changed it to `typemax(Int32)`. 

(It’s weird that this doesn’t seem to be documented anywhere.)

**Comment by stevengj** [comment]
Wow, green CI!

**Comment by stevengj** [comment]
merge?

**Comment by vtjnash** [comment]
Actually, I cannot conclusively prove the kernel avoids negative values, though I would still think the kernel developers would be crazy to allocate them. The limitation on having code use comparisons with `< 0` or `== -1`, as would generally be standard on unix, doesn't work on Win32 because the type there is defined as being unsigned. Awkwardly, as some links point out, the kernel API is a bit of a mess, it is the special value `0` or `NULL` that is actually the one that is returned in half of functions as the invalid value and is actually invalid, while `-1` is actually the bit pattern of the valid handle known as `INVALID_HANDLE_VALUE`, `INVALID_SOCKET`, but is also the valid pseudo-handle for the current window / current process / current thread among others.

And I have noticed that the kernel internally truncates the argument value handles, which seems certainly also an undocumented behavior.

**Pull request merged**